source/add-jupyter/modify-submit-parameters.rst

```diff
@@ -0,0 +1,169 @@
+
+.. code-block:: yaml
+
+   # submit.yml.erb
+   ---
+
+   batch_connect
+     template: "basic"
+
+Which only describes which internal template to use when generating the batch
+script. From here we can add more options to the batch script, such as:
+
+.. code-block:: yaml
+
+   # submit.yml.erb
+   ---
+
+   batch_connect
+     template: "basic"
+     set_host: "host=$(hostname -A | awk '{print $2}')"
+
+where we override the Bash script used to determine the host name of the
+compute node from within a running batch job.
+
+You can learn more about possible ``batch_connect`` options here:
+
+http://www.rubydoc.info/gems/ood_core/OodCore%2FBatchConnect%2FTemplate:initialize
+
+.. warning::
+
+   It is recommended any global ``batch_connect`` attributes be defined in the
+   corresponding cluster configuration file under::
+
+     /etc/ood/config/clusters.d/cluster.yml
+
+   There is further discussion on this under
+   :ref:`install-desktops-modify-cluster-configuration`.
+
+But in most cases you will want to change the actual job submission parameters (e.g., node type). These are defined under the ``script`` option as:
+
+.. code-block:: yaml
+
+   # submit.yml.erb
+   ---
+
+   batch_connect
+     template: "basic"
+
+   script:
+     ...
+
+You can read more about all the available ``script`` options here:
+
+http://www.rubydoc.info/gems/ood_core/OodCore/Job/Script
+
+Although in most cases you will want to modify the ``native`` attribute, which
+is resource manager dependent. Some examples are given below.
+
+.. note::
+
+   It is recommended you commit the changes you made to ``submit.yml.erb`` to
+   `git`_:
+
+   .. code-block:: sh
+
+      # Stage the modified file
+      git add submit.yml.erb
+
+      # Commit your changes
+      git commit -m 'updated batch job options'
+
+Slurm
+`````
+
+For Slurm, you can choose the features on a requested node with:
+
+.. code-block:: yaml
+
+   # submit.yml.erb
+   ---
+
+   batch_connect
+     template: "basic"
+
+   script:
+     native: [ "-N", "<%= bc_num_slots.blank? ? 1 : bc_num_slots.to_i %>", "-C", "c12" ]
+
+where we define the :command:`sbatch` parameters as an array under ``script`` and
+``native``.
+
+.. note::
+
+   The ``bc_num_slots`` shown above located within the ERB syntax is the value
+   returned from web form for "Number of nodes". We check if it is blank and
+   return a valid number.
+
+Torque
+``````
+
+For Torque, you can choose processors-per-node with:
+
+.. code-block:: yaml
+
+   # submit.yml.erb
+   ---
+
+   batch_connect
+     template: "basic"
+
+   script:
+     native:
+       resources:
+         nodes: "<%= bc_num_slots.blank? ? 1 : bc_num_slots.to_i %>:ppn=28"
+
+.. note::
+
+   The ``bc_num_slots`` shown above located within the ERB syntax is the value
+   returned from web form for "Number of nodes". We check if it is blank and
+   return a valid number.
+
+Other
+`````
+
+For most of our adapters (aside from Torque) the ``native`` attribute is an
```

Comment:
You should mention the purpose of this array further up, before the torque and slurm examples.

It's not clear right now what the array does in those cases, or where it gets plugged into the template. It's only when you get to here that it makes sense.